runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/StringSetData.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.core.metrics;
+
+import com.google.auto.value.AutoValue;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.beam.sdk.metrics.StringSetResult;
+
+/**
+ * Data describing the StringSet. This should retain enough detail that it can be combined with
+ * other {@link StringSetData}.
+ */
+@AutoValue
+public abstract class StringSetData implements Serializable {
+
+  public abstract Set<String> stringSet();
```

Comment:
Note: We ended up changing StringSetData  to be mutable because in certain IOs such as TextIOs the number lineage records (per file) are very large and immutable hashset copy cost became huge 

https://github.com/apache/beam/pull/32650/files#diff-5bbbc1e2641f65bfb9040d37f09cddb696d642e7a0bb5f4779b996350141b9ba